pkg/controller/infinispan/resources/config/config_controller.go

```diff
@@ -74,6 +75,16 @@ func Add(mgr manager.Manager) error {
 }
 
 func (c *configResource) Process() (reconcile.Result, error) {
+	// Don't update the ConfigMap if an update will be required
+	if req, err := ispnctrl.IsUpgradeRequired(c.infinispan); req || err != nil {
+		return reconcile.Result{RequeueAfter: consts.DefaultWaitOnCreateResource}, nil
+	}
+
+	// If an upgrade is in progress wait for it to complete before updating the ConfigMap
+	if c.infinispan.IsAnyCondititionTrue(ispnv1.ConditionUpgrade, ispnv1.ConditionStopping) {
```

Comment:
If #763 wouldn't be fixed, this code cause continuous loop

Reply:
True. I wasn't aware of that issue. That issue needs to be fixed eventually, but I don't think it should stop this fix. Worst case scenario the ConfigMap retains the old data whilst `ConditionStopping` and the config-controller will loop until this is resolved.

Reply:
Probably, we could validate upgrade condition again here, i.e. no Pods or Pods with DEFAULT_IMAGE different from operator configuration

Reply:
That's handled in the call to `if req, err := ispnctrl.IsUpgradeRequired(c.infinispan); req || err != nil {`

Reply:
Why we need additional validation ` c.infinispan.IsAnyCondititionTrue` in this case?

Reply:
They handle two different scenarios. The first if statement is to guard against the config-controller processing Infinispan before infinispan_controller can initiate the upgrade and sets the appropriate conditions. The second if statement ensures that if a upgrade has been initiated, we don't update the Config until after the original cluster pods have been shutdown.

Reply:
I'm not sure, but seems to new Pods never will have new DEFAULT_IMAGE value till upgrade has finished. Seems to second validation is bit redundant, but maybe I'm wrong

Reply:
> Seems to second validation is bit redundant, but maybe I'm wrong

Hmm actually I think you're right, as `IsUpgradeRequired` already checks if `ConditionUpgrade` is true. I'll try removing and verify locally.

Reply:
Yes, `ispnv1.ConditionStopping` a bit danger validation as it's related to graceful shutdown also. But in this case we can free to change any config, as it's not upgrade

Reply:
Do we use graceful shutdown for anything other than upgrades?

Reply:
Yes, sure. If anyone defines replicas=0 operator will execute graceful shutdown

Reply:
Ok, in that case it should be fine to remove the `ConditionUpgrade` check as scaling the cluster back up will contain the same Infinispan version and so we can't run into this issue.